### PR TITLE
Added states for endgame, note state, and placement mode

### PR DESCRIPTION
**Pull request by Malachi-lewis**
The rest of the robot will likely need states, so we should have them in RobotStateManager as quickly as possible.

### CODE REVIEW

File: src/main/java/frc/robot/NoteState.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package frc.robot;
+
+import java.util.Arrays;
+
+public enum NoteState {
+  NO_TE(0),
+  LOADING(10),
+  CHAMBER(20),
+  READY_TO_FIRE(30),
```
Comment (lohengrin332) [inline src/main/java/frc/robot/NoteState.java:9]: Technically, `READY_TO_FIRE` is about more than just the note; it's about the note plus the angle of the turret. I would add this as an additional queryable state or boolean.

File: src/main/java/frc/robot/stateManagement/NoteState.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package frc.robot.stateManagement;
+
+import java.util.Arrays;
+
+public enum NoteState {
+  NO_TE(0),
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/stateManagement/NoteState.java:6]: What does this state mean?  I assume no note, let's make it more clear.

File: src/main/java/frc/robot/stateManagement/RobotStateManager.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  public RobotStateManager() {}
+
+  @Override
+  public void periodic() {
+    if (checkAllianceColor) {
+      Optional<Alliance> alliance = DriverStation.getAlliance();
+      if (alliance.isPresent()) {
+        checkAllianceColor = false;
+        alliancePublisher.accept(alliance.get().equals(Alliance.Red) ? 10 : 20);
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/stateManagement/RobotStateManager.java:49]: We should use the enum here instead of 10/20

Comment (Malachi-lewis) [reply src/main/java/frc/robot/stateManagement/RobotStateManager.java:49]: done

File: src/main/java/frc/robot/stateManagement/RobotStateManager.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.util.Optional;
+
+public class RobotStateManager extends SubsystemBase {
+  // Alliance Color
+  private final IntegerTopic allianceTopic =
+      NetworkTableInstance.getDefault().getIntegerTopic("ALLIANCE");
+  private final IntegerPublisher alliancePublisher = allianceTopic.publish();
+  private boolean checkAllianceColor = true;
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/stateManagement/RobotStateManager.java:19]: FMS already stores the result of isRedAlliance.  I don't think we need to have another network table for alliance.  

I would expect RobotStateManager to check whether isRedAlliance in initialization, and store that in a private boolean and then return that information using a getter.

Comment (Malachi-lewis) [reply src/main/java/frc/robot/stateManagement/RobotStateManager.java:19]: done

File: src/main/java/frc/robot/stateManagement/NoteState.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  LOADING(10),
+  CHAMBER(20),
+  READY_TO_FIRE(30),
+  AMP(40);
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/stateManagement/NoteState.java:10]: What does AMP mean here?  Ready for amp placement?  Might want to follow the READY_TO_FIRE pattern, and put READY_TO_ in front of amp.

**Review by caseyjbrotherton — CHANGES_REQUESTED** [review]
Left comments.  Another thought is whether the enums should be in their own class, or if they should be included in RobotStateManagement

We should think more about that.  

Also, another thought.  Do we expect the different subsystems to get the state updates directly from networktables, or from RobotStateManagement?  I assume networktables because we have a NT publisher.  But wanted to be sure.  We will need to make sure to show the team an example so that they follow the same pattern

File: src/main/java/frc/robot/stateManagement/NoteState.java (hunk 5 above)
Comment (caseyjbrotherton) [reply src/main/java/frc/robot/stateManagement/NoteState.java:10]: Talking to Malachi - READY_TO_PLACE?

File: src/main/java/frc/robot/stateManagement/RobotStateManager.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            alliance.get().equals(Alliance.Red) ? AllianceColor.RED : AllianceColor.BLUE;
+      }
+    }
+    if (!isEndGame) {
+      double matchTimer = DriverStation.getMatchTime();
+      boolean isTeleopEnabled = DriverStation.isTeleopEnabled();
+      if (matchTimer < 20 && isTeleopEnabled) {
+        isEndGame = true;
+      }
+    }
```
Comment (azvanderpas) [inline src/main/java/frc/robot/stateManagement/RobotStateManager.java:38]: Consider writing this using [Triggers](https://docs.wpilib.org/en/stable/docs/software/commandbased/binding-commands-to-triggers.html#arbitrary-triggers)

**Review by caseyjbrotherton — DISMISSED** [review]
Changes look good

**Review by caseyjbrotherton — DISMISSED** [review]

**Review by caseyjbrotherton — APPROVED** [review]

File: src/main/java/frc/robot/stateManagement/RobotStateManager.java (hunk 6 above)
Comment (caseyjbrotherton) [reply src/main/java/frc/robot/stateManagement/RobotStateManager.java]: Trigger added

**Review by caseyjbrotherton — APPROVED** [review]

File: src/main/java/frc/robot/NoteState.java (hunk 1 above)
Comment (caseyjbrotherton) [reply src/main/java/frc/robot/NoteState.java]: Resolving to merge today.  @lohengrin332 let's talk about this on Tuesday and improve forward

**Review by caseyjbrotherton — APPROVED** [review]

**Pull request merged**